WalletWasabi/Microservices/ProcessInvoker.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace WalletWasabi.Microservices
+{
+	public class ProcessInvoker
+	{
+		public async Task<(string response, int exitCode)> SendCommandAsync(ProcessStartInfo startInfo, CancellationToken token, Action<StreamWriter>? standardInputWriter = null)
+		{
+			using var processAsync = new ProcessAsync(startInfo);
+
+			if (standardInputWriter is { })
+			{
+				processAsync.StartInfo.RedirectStandardInput = true;
+			}
+
+			processAsync.Start();
+
+			if (standardInputWriter is { })
+			{
+				standardInputWriter(processAsync.StandardInput);
+				processAsync.StandardInput.Close();
+			}
```

Comment:
Isn't this easier to read? and we check `standardInputWriter` only once:
```suggestion
			if (standardInputWriter is null)
			{
				processAsync.Start();
			}
			else
			{
				processAsync.StartInfo.RedirectStandardInput = true;
				processAsync.Start();
				standardInputWriter(processAsync.StandardInput);
				processAsync.StandardInput.Close();
			}
```

Reply:
I'm neutral on this. If others like it, why not.

Reply:
I prefer the original one too.

Reply:
My suggested code is not only stylish change, the original code checks `standardInputWriter` twice unnecessarily.

Reply:
> My suggested code is not only stylish change, the original code checks `standardInputWriter` twice unnecessarily.

CLR has quite powerful JIT compiler that may re-order commands, do some optimizations etc. So I dare to say it is a stylish change. 

I don't like the suggestion because it repeats `processAsync.Start()` and it kind of suggest to me that there are two distinct operations that the method can do.  But in reality, `standardInputWriter` only affects how it does the one thing it is designed to do.

But I think we have gotten to bikeshedding here.